File: server/dotnet/Controllers/Render.cs
using System.Text;
using Microsoft.AspNetCore.Mvc;

public class RenderPaypalController : Controller
{
    private readonly TemplatePathResolver _templatePathResolver;
    private readonly IHttpClientFactory _httpClientFactory;

    public RenderPaypalController(TemplatePathResolver templatePathResolver, IHttpClientFactory httpClientFactory)
    {
        _templatePathResolver = templatePathResolver;
        _httpClientFactory = httpClientFactory;
    }

    [HttpGet("/")]
    public async Task<IActionResult> PaypalRender()
    {
        var isFlexibleIntegration = false;
        var queryParams = HttpContext.Request.Query;

        foreach (var param in queryParams)
        {
            if (param.Key == "flexible")
            {
                isFlexibleIntegration = true;
            }
        }

        var sdkUrl = SdkScriptHelpers.GetPayPalSdkUrl();
        var clientToken = await SdkScriptHelpers.GetClientTokenAsync(_httpClientFactory);

        var locals = new Dictionary<string, string>
        {
            { "title", "Fastlane - Paypal Integration" + (isFlexibleIntegration ? " (Flexible)" : "") },
            { "prerequisiteScripts", $@"
                <script
                    src=""{sdkUrl}""
                    data-sdk-client-token=""{clientToken}""
                    defer
                ></script>
            " },
            { "initScriptPath", isFlexibleIntegration ? "init-fastlane-flexible.js" : "init-fastlane.js" },
            { "stylesheetPath", "styles.css" }
        };

        var renderedHtml = await _templatePathResolver.RenderTemplateAsync(isFlexibleIntegration, locals);

        return Content(renderedHtml, "text/html", Encoding.UTF8);
    }
}

File: server/python/controllers/render.py
import os
import chevron
from flask import make_response

from helpers import get_paypal_sdk_url, get_client_token


def render_paypal(args, templates_folder):
    is_flexible_integration = "flexible" in args

    sdk_url = get_paypal_sdk_url()
    client_token = get_client_token()

    locals = {
        "title": "Fastlane - PayPal Integration"
        + (" (Flexible)" if is_flexible_integration else ""),
        "prerequisiteScripts": f"""
            <script
                src="{sdk_url}"
                data-sdk-client-token="{client_token['clientToken']}"
                defer
            ></script>
        """,
        "initScriptPath": (
            "init-fastlane-flexible.js"
            if is_flexible_integration
            else "init-fastlane.js"
        ),
        "stylesheetPath": "styles.css",
    }

    template_name = (
        "checkout-flexible.html" if is_flexible_integration else "checkout.html"
    )
    template_path = os.path.join(templates_folder, template_name)

    with open(template_path, "r") as f:
        template = f.read()

    html = chevron.render(template, locals)
    response = make_response(html)
    return response
